### PR TITLE
Update gtest deps

### PR DESCRIPTION
**Pull request by streeve**
Replace #11 - packages only install source. Instead build and install directly

Still related to https://github.com/ECP-copa/Cabana/pull/510

### CODE REVIEW

File: fedora [hunk 1]
```diff
@@ -4,7 +4,25 @@ FROM ghcr.io/kokkos/ci-containers/fedora:${TAG}
...
 RUN ( dnf -y update || dnf -y update ) && \
     dnf -y install \
-      openmpi-devel fftw-devel doxygen valgrind libasan gtest && \
+      openmpi-devel fftw-devel doxygen valgrind libasan && \
     dnf clean all
+
+# Install Googletest
+ENV GTEST_DIR=/opt/gtest
+RUN GTEST_VERSION=release-1.11.0 && \
+    GTEST_URL=https://github.com/google/googletest/archive/${GTEST_VERSION}.tar.gz && \
```
Comment (dalg24) [inline fedora]:
```suggestion
ARG GTEST_VERSION=release-1.11.0
RUN GTEST_URL=https://github.com/google/googletest/archive/${GTEST_VERSION}.tar.gz && \
```

Comment (streeve) [reply fedora]: Fixed

**Review by junghans — COMMENTED** [review]
Is this needed for all distros?

**Comment by streeve** [comment]
> Is this needed for all distros?

We run tests on all distros, so I think so

**Comment by junghans** [comment]
> > Is this needed for all distros?
> 
> We run tests on all distros, so I think so

I meant is only the ubuntu package broken or all of them?

**Comment by streeve** [comment]
> I meant is only the ubuntu package broken or all of them?

Oh, yes it's the same behavior for all of them

**Comment by junghans** [comment]
> > I meant is only the ubuntu package broken or all of them?
> 
> Oh, yes it's the same behavior for all of them

Something is wrong here...Fedora (at least) install all the gtest headers and libraries:
https://src.fedoraproject.org/rpms/gtest/blob/rawhide/f/gtest.spec#_81

**Comment by junghans** [comment]
As for opensuse: https://build.opensuse.org/package/view_file/openSUSE:Factory/googletest/googletest.spec?expand=1

Bottom line: we should try to make our CMake detect and use gtest from those packages....

**Comment by junghans** [comment]
Hmm, it seems the packages come without the cmake config file...

**Comment by junghans** [comment]
Ok for Fedora we have to wait for Fedora 36, which has gest-1.11 (Fedora 35 still ships gtest-1.10).

On OpenSUSE the CMake files for `gtest` are in the `gmock` package !?

**Comment by junghans** [comment]
For ubuntu only Jammy (22.04) will have gtest-1.11.

**Comment by junghans** [comment]
ok, here is what I would do:

Fedora
```
UN source /etc/os-release && if [ "${VERSION_ID}" -gt 35 ]; then \
  ( dnf -y update || dnf -y update ) && \
  dnf -y install gtest && \
  dnf clean all; \
else \
  <build gtest manually>
fi
```

Ubuntu:
```
RUN . /etc/os-release && if  [ -n "${VERSION_ID##22*}" ]; then \
  apt-get update && \
  apt-get install -y getest && \
  apt-get purge --autoremove -y && \
  rm -rf /var/lib/apt/lists/*; \
else \
  <build gtest manually>
fi
```

opensuse install the `gmock` package.

**Review by junghans — APPROVED** [review]

**Comment by junghans** [comment]
I tested `ecp-copa/ci-containers/*:latest`, so this should work now, will merge and re-run https://github.com/ECP-copa/Cabana/pull/510.

**Comment by streeve** [comment]
Thanks @junghans !

**Pull request merged**